modules/fedimint-wallet-client/src/lib.rs

```diff
@@ -127,61 +172,177 @@ impl WalletClientExt for Client {
+                        yield DepositState::Failed(format!("Failed to claim: {e:?}"));
+                        return;
+                    }
+
+                    if let Some(out_point) = claiming.change.as_ref() {
+                        self.await_primary_module_output(operation_id, *out_point)
+                            .await
+                            .expect("Cannot fail if tx was accepted and federation is honest");
+                    }
+                    yield DepositState::Claimed;
                 }
-                Some(s) => {
-                    panic!("Unexpected state {s:?}")
-                },
-                None => return,
-            }
+            }),
+        )
+    }
 
-            match next_deposit_state(&mut operation_stream).await {
-                Some(DepositStates::WaitingForConfirmations(_)) => {
-                    yield DepositState::WaitingForConfirmation;
-                },
-                Some(s) => {
-                    panic!("Unexpected state {s:?}")
-                },
-                None => return,
-            }
+    async fn get_withdraw_fee(
+        &self,
+        address: Address,
+        amount: bitcoin::Amount,
+    ) -> anyhow::Result<PegOutFees> {
+        let (wallet_client, _) =
+            self.get_first_module::<WalletClientModule>(&WalletCommonGen::KIND);
 
-            let claiming = match next_deposit_state(&mut operation_stream).await {
-                Some(DepositStates::Claiming(claiming)) => claiming,
-                Some(s) => {
-                    panic!("Unexpected state {s:?}")
-                },
-                None => return,
-            };
-            yield DepositState::Confirmed;
+        wallet_client.get_withdraw_fees(address, amount).await
+    }
 
-            if let Err(e) = tx_subscriber.await_tx_accepted(claiming.transaction_id).await {
-                yield DepositState::Failed(format!("Failed to claim: {e:?}"));
-                return;
-            }
+    async fn withdraw(
+        &self,
+        address: Address,
+        amount: bitcoin::Amount,
+        fee: PegOutFees,
+    ) -> anyhow::Result<OperationId> {
+        let (wallet_client, instance) =
+            self.get_first_module::<WalletClientModule>(&WalletCommonGen::KIND);
 
-            if let Some(out_point) = claiming.change.as_ref() {
-                self.await_primary_module_output(operation_id, *out_point)
-                    .await
-                    .expect("Cannot fail if tx was accepted and federation is honest");
-            }
-            yield DepositState::Claimed;
-        }))
+        let operation_id = OperationId(thread_rng().gen());
+
+        let withdraw_output = wallet_client
+            .create_withdraw_output(operation_id, address.clone(), amount, fee.clone())
+            .await?;
+        let tx_builder =
+            TransactionBuilder::new().with_output(withdraw_output.into_dyn(instance.id));
```

Comment:
Is it possible for this transaction to create change?

Reply:
Yes, it is, you think we should wait for that to arrive before considering the operation finished for easier testing?

Reply:
Ideally I think the outer state machine should have a state that awaits the change, although I see that the devimint test doesn't directly verify the change in ecash notes so maybe its not necessary

Reply:
I added rudimentary change awaiting, not sure how useful this is for real apps, but ok with me for now since it allowed me to also test the balance change.